go/vt/mysqlctl/mysqld.go

```diff
@@ -1176,9 +1184,20 @@ func buildLdPaths() ([]string, error) {
 	return ldPaths, nil
 }
 
-// GetVersionString is part of the MysqlDaemon interface.
-func (mysqld *Mysqld) GetVersionString() string {
-	return fmt.Sprintf("%d.%d.%d", mysqld.capabilities.version.Major, mysqld.capabilities.version.Minor, mysqld.capabilities.version.Patch)
+// GetVersionString is part of the MysqlExecutor interface.
+func (mysqld *Mysqld) GetVersionString(ctx context.Context) string {
+	qr, err := mysqld.FetchSuperQuery(ctx, "select @@global.version")
+	if err != nil {
+		log.Errorf("Error fetching MySQL version: %v", err)
+		return ""
+	}
+	if len(qr.Rows) != 1 {
+		log.Errorf("Unexpected number of rows: %v", qr.Rows)
+		return ""
+	}
+	res := qr.Named().Row()
```

Comment:
BTW the function `Row()` returns `nil` if the number of rows `!= 1`. So if you wanted, you could rewrite the integrity check above from `if len(qr.Rows) != 1 {` to (down in this line) `if res == nil`. Whichever you feel more comfortable with, seeing that the former is more explicit and the latter is implied.

Reply:
Same here, this is basically a copy paste of how the comment version works and mostly the same to stay consistent with that.